workshops/pycuriosity/README.md

```diff
@@ -0,0 +1,417 @@
+PyCuriosity is an API explorer that gets Curiosity's images and image urls. 
+
+If you haven't worked with APIs before, read this [resource](https://www.ibm.com/in-en/topics/api). In short, 
+**A**pplication **P**rogramming **I**nterfaces(APIs for short) enable different computer applications to communicate, 
+which is useful because we want images from Curiosity, which are stored on NASA's servers. To access them, we need a way 
+to communicate with NASA's servers, which may seem daunting, but all we need to do is make a [GET request](https://rapidapi.com/blog/api-glossary/get)
+to one of NASA's APIs, or request information from NASA's servers.
+
+You can browse through NASA's APIs at [api.nasa.gov](https://api.nasa.gov/), and sign up for an API key if you want.
+This isn't strictly required, as we'll see later on. The API we'll be using in this workshop is called the [Mars Rover Photos API](https://github.com/corincerami/mars-photo-api),
+an API for, well, Mars rover photos.
+
+For simplicity, we won't create a GUI, but rather enable program execution using command line arguments(arguments that
+can modify the flow of a program), using Python's builtin ```argparse```, a powerful library to parse command line arguments.
+
+After validating the command line arguments, we'll call the Mars Rover Photos API, using ```requests```, and then write
+the data to a text file, and potentially save the first image if we find any.
+
+## The Plan
+Be sure to read the previous section if you're still confused as to what PyCuriosity is.
+
+All great projects, big or small, always start with a plan. Planning out the project ahead of time saves confusion and 
+provides a clear picture of the project, the resources to use, the skills to acquire. 
+
+![Battle Plan GIF from Home Alone](https://cloud-odfqnnmkf-hack-club-bot.vercel.app/0giphy.gif)
+
+This program flow was made using [diagrams.net](https://app.diagrams.net), be sure to check that out if you're planning
+on making a big project, especially with a team.
+
+![PyCuriosity program flow](https://cloud-2o6nek83e-hack-club-bot.vercel.app/0curiosity_battle_plan.png)
+
+## Init
+First, we need to look at the Mars Rover Photos API [documentation](https://github.com/corincerami/mars-photo-api), and 
+it's a good idea to look it up on [api.nasa.gov](https://api.nasa.gov/) as well. We'll be querying by Earth date to keep things simple.
+
+After going through the documentation, the following parameters are necessary to include in the request(take a look at the sample query)-
+1. An API key
+2. The Earth date(date Curiosity clicked the photos), which we need to get from the user
+
+Also, we need a place to store the data we're requesting, so we also need a file name from the user.
+
+#### Starting the project
+Fork the [starting repl](https://replit.com/@agentblack6000/pycuriosity-init?v=1). Let's see what we got-
+- Right away you'll see a bunch of TODOs, incase you want to attempt the project solo
```

Comment:
the TODOs thing is very cool hehe! i'm a little confused by these documentation TODOs though...
<img width="301" alt="Screenshot 2023-02-17 at 1 01 41 PM" src="https://user-images.githubusercontent.com/65808924/219553578-acd6e7b0-e6f5-463c-8f5d-50fc8df02dd7.png">

Reply:
ack i got to the end of the workshop and just realised what this means, ignore me haha